wire-runtime/src/test/java/com/squareup/wire/protos/foreign/ForeignEnum.java

```diff
@@ -2,22 +2,51 @@
 // Source file: ../wire-runtime/src/test/proto/foreign.proto at 23:1
 package com.squareup.wire.protos.foreign;
 
-import com.squareup.wire.ProtoEnum;
+import com.squareup.wire.ProtoReader;
+import com.squareup.wire.ProtoWriter;
+import com.squareup.wire.TypeAdapter;
+import java.io.IOException;
+import java.lang.IllegalStateException;
 import java.lang.Override;
 
-public enum ForeignEnum implements ProtoEnum {
+public enum ForeignEnum {
   BAV(0),
 
   BAX(1);
 
-  private final int value;
+  public static final TypeAdapter<ForeignEnum> ADAPTER = new TypeAdapter.EnumAdapter<ForeignEnum>() {
+    @Override
+    public int serializedSize(ForeignEnum value) {
+      return TypeAdapter.UINT32.serializedSize(value.value);
+    }
+
+    @Override
+    public void write(ForeignEnum value, ProtoWriter writer) throws IOException {
+      TypeAdapter.UINT32.write(value.value, writer);
+    }
+
+    @Override
+    public ForeignEnum read(ProtoReader reader) throws IOException {
+      int value = TypeAdapter.UINT32.read(reader);
+      ForeignEnum constant = fromValue(value);
+      if (constant == null) {
+        throw new IllegalStateException("No constant for value " + value);
+      }
+      return constant;
+    }
+  };
+
+  public final int value;
 
   ForeignEnum(int value) {
     this.value = value;
   }
 
-  @Override
-  public int getValue() {
-    return value;
+  public static ForeignEnum fromValue(int value) {
```

Comment:
`fromValue` needs to be `static` since it's used without an instance. I originally had `TypeAdapter.ENUM` for all enums which did this very thing.